### PR TITLE
Trusted entitlements

### PR DESCRIPTION
**Pull request by tonidero**
### Description
This new feature prevents MitM attacks between the SDK and the RevenueCat server.
With verification enabled, the SDK ensures that the response created by the server was not modified by a third-party, and the entitlements received are exactly what was sent.
This is 100% opt-in. `EntitlementInfos` have a new `VerificationResult` property, which will indicate the validity of the responses when this feature is enabled.

This feature is experimental and requires opt in. It's also only available in Kotlin.

```kotlin
@OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
fun configureRevenueCat() {
    val configuration = PurchasesConfiguration.Builder(context, apiKey)
        .informationalVerificationModeAndDiagnosticsEnabled(true)
        .build()
    Purchases.configure(configuration)
}
```

### CODE REVIEW

**Comment by tonidero** [comment]
Will hold this in a branch until the next release is done but this is ready for review

File: api-tester/src/main/java/com/revenuecat/apitester/java/EntitlementInfosAPI.java [hunk 1]
```diff
@@ -11,19 +12,16 @@ static void check(final EntitlementInfos infos) {
...
-        // final VerificationResult verification = infos.getVerification();
+        final VerificationResult verification = infos.getVerification();
     }
 
     static void checkConstructor(
-            Map<String, EntitlementInfo> all
-            // Trusted entitlements: Commented out until ready to be made public
-            // VerificationResult verificationResult
+            Map<String, EntitlementInfo> all,
+             VerificationResult verificationResult
```
Comment (vegaro) [inline api-tester/src/main/java/com/revenuecat/apitester/java/EntitlementInfosAPI.java:20]:
```suggestion
            VerificationResult verificationResult
```

File: common/src/main/java/com/revenuecat/purchases/common/EntitlementInfoFactories.kt [hunk 2]
```diff
@@ -76,7 +75,6 @@ internal fun JSONObject.optOwnershipType(name: String) = when (optString(name))
     else -> OwnershipType.UNKNOWN
 }
 
-// Trusted entitlements: Added until ready to use it.
 @Suppress("UnusedPrivateMember")
```
Comment (vegaro) [inline common/src/main/java/com/revenuecat/purchases/common/EntitlementInfoFactories.kt:78]: can this be removed now?

**Comment by tonidero** [comment]
Holding until #1061 and #1066 are merged into this so it can be merged as a whole.

**Pull request merged**